chips/rp2040/src/cyw43439/utils.rs

```diff
@@ -0,0 +1,185 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2025.
+
+use core::slice;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+use kernel::utilities::registers::{register_bitfields, FieldValue, LocalRegisterCopy};
+
+enum_from_primitive! {
+#[derive(Debug, Clone, Copy)]
+pub enum ChannelType {
+    Control = 0,
+    Event = 1,
+    Data = 2
+}
+}
+
+pub(super) fn as_bytes<T: Sized>(data: &T) -> &[u8] {
+    unsafe {
+        core::slice::from_raw_parts(
+            core::ptr::from_ref::<T>(data) as *const u8,
+            core::mem::size_of::<T>(),
+        )
+    }
+}
+
+pub(super) fn slice8_mut(x: &mut [u32]) -> &mut [u8] {
+    let len = x.len() * 4;
+    unsafe { slice::from_raw_parts_mut(x.as_mut_ptr() as _, len) }
+}
+
```

Comment:
With these unsafes I don't know how necessary they are, but it would be good to add a `# Safety` comment that explains why they are safe.